### PR TITLE
Fixes for Comment Issues

### PR DESCRIPTION
**Pull request by dshanske**
* Update Avatar Code to not override local user data
* Add support for storing the author URL in proper place
* Turn whitelist into blacklist for mf2 properties
* Sanitize mf2 url properties
* No longer store source in two places if webmention
* Remove merge between slashed and unslashed data as 4.7.1 update now slices array after filter
* Remove compatibility function and require 4.7.1 minimum
* Only run function on update if webmention and no longer try to update other linkbacks due no protocol support

### CODE REVIEW

File: includes/class-linkbacks-handler.php [hunk 1]
```diff
@@ -423,11 +427,21 @@ public static function get_comment_link( $link, $comment, $args ) {
...
 	public static function get_comment_author_url( $url, $id, $comment ) {
-		if ( $author_url = get_comment_meta( $id, 'semantic_linkbacks_author_url', true ) ) {
-			return $author_url;
+		$author_url = get_comment_meta( $id, 'semantic_linkbacks_author_url', true );
+		if ( ! $author_url ) {
+			return $url;
 		}
-
-		return $url;
+		if ( get_comment_meta( $id, 'webmention_source_url' ) ) {
```
Comment (pfefferle) [inline includes/class-linkbacks-handler.php]: I am not sure if this is a good place... we should not change data in a "getter" function...

File: includes/class-linkbacks-handler.php [hunk 2]
```diff
@@ -81,14 +81,17 @@ public static function register_meta() {
 	 * Update an Enhanced Comment
 	 */
 	public static function enhance( $commentdata, $comment = array(), $commentarr = array() ) {
-		if ( ! empty( $commentarr ) ) {
-			// add pre-processed data from, for example the Webmention plugin
-			$commentdata = array_merge( $commentdata, $commentarr );
+		// If $comment is empty this is a new comment
```
Comment (pfefferle) [inline includes/class-linkbacks-handler.php]: is this a bug that came with the enhance function?

File: includes/class-linkbacks-mf2-handler.php [hunk 3]
```diff
@@ -192,7 +192,12 @@ public static function generate_commentdata( $commentdata ) {
 				}
 
 				if ( isset( $author['url'] ) ) {
-					$commentdata['comment_meta']['semantic_linkbacks_author_url'] = $author['url'];
+					if ( in_array( $commentdata['comment_type'], array( 'webmention', '' ) ) ) {
```
Comment (pfefferle) [inline includes/class-linkbacks-mf2-handler.php]: if we replace the author_url we break the comment detection/duplicate comment code of the web mention plugin: https://github.com/pfefferle/wordpress-webmention/blob/master/includes/class-webmention-receiver.php#L457

**Comment by dshanske** [comment]
I think the meta key location storage issue has to be thought out a bit more then. I could do a temporary fix to fix the existing problem which people are having issues with, giving us more time to address this. That is probably the best move.

**Comment by dshanske** [comment]
Going to close in favor of a temporary fix for the problem and will address the bigger problem later. Need to fix the bugs.

**Pull request closed without merging**